src/main/java/com/google/collegeplanner/servlets/DatastoreServlet.java

```diff
@@ -87,18 +90,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         // Successful. The default status code response is 200.
         return;
       }
-      addCourses(coursesArray, response);
+
+      try {
+        addCourses(coursesArray);
+      } catch (IllegalArgumentException e) {
+        respondWithError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
```

Comment:
should this also `continue` instead of error...? i feel like the same logic from your other comment would apply here.